WooCommerce/WooCommerceTests/ViewRelated/CardPresentPayments/CardPresentPaymentsModalViewControllerTests.swift

```diff
@@ -0,0 +1,155 @@
+import XCTest
+@testable import WooCommerce
+
+final class CardPresentPaymentsModalViewControllerTests: XCTestCase {
+
+    func test_viewcontroller_presents_top_title_provided_by_viewmodel() throws {
+        let viewModel = ModalViewModel(mode: .fullInfo)
+        let viewController = CardPresentPaymentsModalViewController(viewModel: viewModel)
+
+        _ = try XCTUnwrap(viewController.view)
+
+        XCTAssertEqual(viewController.getTopTitleLabel().text, viewModel.topTitle)
+    }
+
+    func test_viewcontroller_presents_top_subtitle_provided_by_viewmodel() throws {
+        let viewModel = ModalViewModel(mode: .fullInfo)
+        let viewController = CardPresentPaymentsModalViewController(viewModel: viewModel)
+
+        _ = try XCTUnwrap(viewController.view)
+
+        XCTAssertEqual(viewController.getTopSubtitleLabel().text, viewModel.topSubtitle)
+    }
+
+    func test_viewcontroller_presents_image_provided_by_viewmodel() throws {
+        let viewModel = ModalViewModel(mode: .fullInfo)
+        let viewController = CardPresentPaymentsModalViewController(viewModel: viewModel)
+
+        _ = try XCTUnwrap(viewController.view)
+        let image = viewController.getImageView().image
+
+        XCTAssertEqual(image, viewModel.image)
+    }
+
+    func test_viewcontroller_presents_bottom_title_provided_by_viewmodel() throws {
+        let viewModel = ModalViewModel(mode: .fullInfo)
+        let viewController = CardPresentPaymentsModalViewController(viewModel: viewModel)
+
+        _ = try XCTUnwrap(viewController.view)
+
+        XCTAssertEqual(viewController.getBottomTitleLabel().text, viewModel.bottomTitle)
+    }
+
+    func test_viewcontroller_presents_bottom_subtitle_provided_by_viewmodel() throws {
+        let viewModel = ModalViewModel(mode: .fullInfo)
+        let viewController = CardPresentPaymentsModalViewController(viewModel: viewModel)
+
+        _ = try XCTUnwrap(viewController.view)
+
+        XCTAssertEqual(viewController.getBottomSubtitleLabel().text, viewModel.bottomSubtitle)
+    }
+
+    func test_viewcontroller_propagates_tap_in_primary_button_to_viewmodel() throws {
```

Comment:
For more coverage, you may want to test primary button against .oneActionButton and .reducedInfoOneActionButton